porch/pkg/engine/engine.go

```diff
@@ -265,9 +265,11 @@ func (cad *cadEngine) UpdatePackageRevision(ctx context.Context, repositoryObj *
 			}
 			mutation := &updatePackageMutation{
 				task:              newTask,
+				oldTask:           oldTask,
```

Comment:
still new to this code, do you mind explaining what is `oldTask` and `newTask`?

Reply:
Good question.

So the way a downstream package is created today, is by doing a `clone` operation on an upstream package. `Clone` info is stored in a Task struct that is part of the `Spec` of the PackageRevision API. Now, for updating a downstream package, the plan is to update the Upstream Ref in the existing clone task of the API.

The code that you are looking here gets invoked when PackageRevision is updated by updating the upstream bits in the clone task of the PackageRevision API. New bits are part of the `newTask`. When we are performing an package upgrade (i.e. `updatePackageMutation`), we need the current upstream ref and the target upstream ref. current upstream ref is stored in the current clone task that I am storing as `oldTask`. Now that I think of it, I don't need the whole task but just the upstream info (and that is also just the PackageRevisionRef), I could store just that and that may also be more clear from code/maintenance perspective as well.